conan/tools/cmake/toolchain/blocks.py

```diff
@@ -252,6 +252,27 @@ def context(self):
             return
         return {"arch_flag": arch_flag}
 
+class RpathLinkFlagsBlock(Block):
+    # TODO: should do this only on Linux, or toggled by an opt-in conf
```

Comment:
Probably what we want to do is to enable this directly in the ``conan_cmakedeps_paths.cmake`` generated by the ``CMakeConfigDeps``. That file is  loaded by the ``CMakeToolchain``, but not only for this, so it can be also used by other solutions as ``cmake-conan``, because otherwise whatever we put in the toolchain is not available for ``cmake-conan`` and the CLion-plugin.

So the opt-in would be to use ``CMakeConfigDeps``? It is getting more ready and mature, it is known to solve a lot of problems, so probably it is good to start moving it forward a bit more?